### ISSUE TITLE
cannot build features/0.4.9-alpha on debian stretch

### ISSUE THREAD

**Issue by ftoledo** [M-griffin/EtherTerm#56]
Hi!!
I just testing the latest branch and i cannot build. First i had to install codelite to to "fix" the hardcoded paths in the linux/EtherTerm.mk (ProjectPath and WorkspachePath) then i can start compile but have follow error:

thanks!

```
toledones@gollum:~/src/etherterm/EtherTerm/linux$ make
----------Building project:[ EtherTerm - Debug ]----------
make[1]: se entra en el directorio '/home/toledones/src/etherterm/EtherTerm/linux'
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/input_handler.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_input_handler.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/link_list.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_link_list.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/main.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_main.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/menu_config.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_menu_config.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/menu_function.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_menu_function.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/menu_io.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_menu_io.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/menu_manager.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_menu_manager.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/protocol.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_protocol.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/renderer.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_renderer.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/screen_buffer.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_screen_buffer.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/sequence_decoder.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_sequence_decoder.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/sequence_parser.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_sequence_parser.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/session_manager.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_session_manager.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/surface_manager.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_surface_manager.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/telnet_manager.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_telnet_manager.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/window_manager.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_window_manager.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/io_service.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_io_service.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/socket_handler.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_socket_handler.cpp.o -I. -I. -I.. -I../src 
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/socket_ssh.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_socket_ssh.cpp.o -I. -I. -I.. -I../src 
/home/toledones/src/etherterm/EtherTerm/src/socket_ssh.cpp: In member function ‘int SSH_Socket::verify_knownhost()’:
/home/toledones/src/etherterm/EtherTerm/src/socket_ssh.cpp:332:13: warning: attributes at the beginning of statement are ignored [-Wattributes]
             [[gnu::fallthrough]]; // c++11 and c++14
             ^
/usr/bin/g++  -c  "/home/toledones/src/etherterm/EtherTerm/src/socket_telnet.cpp" -g -O0 -pedantic -W -std=c++11 -Wall  -o ./Debug/up_src_socket_telnet.cpp.o -I. -I. -I.. -I../src 
/home/toledones/src/etherterm/EtherTerm/src/socket_telnet.cpp:21:5: error: ‘SOCKET’ does not name a type
     SOCKET channel;
     ^~~~~~
/home/toledones/src/etherterm/EtherTerm/src/socket_telnet.cpp: In member function ‘virtual int SDL_Socket::getSocketInstance()’:
/home/toledones/src/etherterm/EtherTerm/src/socket_telnet.cpp:167:26: error: ‘struct _TCPsocket’ has no member named ‘channel’
     return m_tcp_socket->channel;
                          ^~~~~~~
EtherTerm.mk:250: fallo en las instrucciones para el objetivo 'Debug/up_src_socket_telnet.cpp.o'
make[1]: *** [Debug/up_src_socket_telnet.cpp.o] Error 1
make[1]: se sale del directorio '/home/toledones/src/etherterm/EtherTerm/linux'
Makefile:4: fallo en las instrucciones para el objetivo 'All'
make: *** [All] Error 2
toledones@gollum:~/src/etherterm/EtherTerm/linux$ 
```

**Comment by ftoledo** [comment]
hardcoded paths follow #54

**Comment by M-griffin** [comment]
> hardcoded paths follow #54

Hi ftoledo, thanks for your interest.  Lets see if we can get you going.  I haven't run Debian in a while, if I remember it always had an older compiler version.  I use Mint for most of my linux development which usually has GCC/G++ 5.1 - 8..   Can you do gcc --version and let me know what version you have currently?

If it's less than 5.1 then you might need to upgrade.  Otherwise, I'll have to take some time to install Debian on a virtual machine and see what the differences are..  I'll get you going though one way or another.

**Comment by M-griffin** [comment]
Took a look at Debian 9 - Stretch, looks like gcc 6 is installed.  So i'll fire it up in a VM and see whats going on.  I'll let you know what I find.

**Comment by ftoledo** [comment]
I found this https://stackoverflow.com/questions/11372036/socket-does-not-name-a-type
and with this patch now compile. I don't known if will affect the funcionality 

```
iff --git a/src/socket_irc.cpp b/src/socket_irc.cpp
index a668e29..7f65ff3 100644
--- a/src/socket_irc.cpp
+++ b/src/socket_irc.cpp
@@ -18,7 +18,7 @@
  */
 struct _TCPsocket {
     int ready;
-    SOCKET channel;
+    int channel;
     IPaddress remoteAddress;
     IPaddress localAddress;
     int sflag;
diff --git a/src/socket_telnet.cpp b/src/socket_telnet.cpp
index 548600a..cc097a3 100644
--- a/src/socket_telnet.cpp
+++ b/src/socket_telnet.cpp
@@ -18,7 +18,7 @@
  */
 struct _TCPsocket {
     int ready;
-    SOCKET channel;
+    int channel;
     IPaddress remoteAddress;
     IPaddress localAddress;
     int sflag;
```

**Comment by M-griffin** [comment]
Thanks for sharing.  
Looks good so far.  Usually a SOCKET is an INT type and usually pre-defined in the include files.

I installed the VM with Debian last night but ran out of time, I'll take a look at this probably late tonight and see if this is good.  or maybe checking 
```
#ifndef SOCKET
#define SOCKET int
#endif
```
If the following is needed.

**Comment by M-griffin** [comment]
This was a few months back, i just compiled it all on Debian Stretch and didn't see the same error.  I'll still mark this as WIP and make the needed change if I forgot to back then.  Thanks again for the report.

**Comment by M-griffin** [comment]
This is corrected with commit:
https://github.com/M-griffin/EtherTerm/commit/bfed14ae9b07033ab36925c972fbd9d530c867aa

On the 0.4.9-alpha branch,  Fixes are being added to the 0.4.10-alpha branch also
